src/module.c

```diff
@@ -6507,6 +6507,49 @@ const char *RM_CallReplyProto(RedisModuleCallReply *reply, size_t *len) {
     return callReplyGetProto(reply, len);
 }
 
+/* Redis only exposes "side effects" of command execution at the end of an execution unit.
+ *
+ * Side effects currently handled are
+ * - propagating replication/AOF operations
+ * - executing the commands of clients blocked on keys modified by commands
+ *
+ * In general, for modules, execution unit start and end with the taking/release of the GIL.  For module code executed
+ * on the Redis main thread, this is done implicitly by Redis itself, while for module code executed on a separate
+ * thread, this is done explicitly by the module itself.
+ *
+ * This function enables modules to flush the execution unit within the middle of an execution unit, thereby having
+ * side effects become visible and impact future operations that occur within the execution unit.
+ *
+ * Usage Example: a module runs multiple commands via RM_Call within a single execution unit, but wants
+ * processCommand() like semantics, where blocked clients that could be unblocked by a previous RM_Call operation will
+ * be executed before the next RM_Call is made.  By calling RM_FlushExecutionUnit between RM_Call operations, the
+ * module will get those semantics.
+ *
+ * The flags is a bit mask of these:
+ *
+ * - `REDISMODULE_FLUSH_EXEC_UNIT_FLAG_DEFAULT`: We propagate pending command operations (AOF/Replication) and unblock
+ *                                               clients
```

Comment:
Instead of listing what operations occur, I would say this is the same as if you sent a multi-exec of commands. If in the future we support new use cases, they should be framed in how they change the consistency of that.

Reply:
ok, i would still like to keep the `flags` argument so we have some flexibility to change things in the future.
can you suggest the text or edit the PR?

Reply:
Sure, I'll suggest the update.